Add Filtering Hands-on to Data Manipulation Olympics.md

lybCNU opened the pull request:
### Description:
This pull request enhances the **Filtering** section of the **Data Manipulation Olympics** tutorial in the Galaxy Training Network (GTN) by adding a new hands-on exercise.

### Key Updates:
- **Added Hands-on Exercise**:
  - Demonstrates the use of Python-based filtering expressions in Galaxy.
  - Includes practical examples such as:
    - Filtering based on substring occurrence.
    - Filtering based on whether values start or end with specific characters.
    - Filtering based on counting occurrences of a substring within columns.

This addition aims to provide users with a more comprehensive understanding of advanced filtering techniques in Galaxy.

shiltemann commented:
Thanks a lot @lybCNU, this is awesome! 

Please add yourself to our [CONTRIBUTORS.yaml](https://github.com/galaxyproject/training-material/blob/main/CONTRIBUTORS.yaml) file, and then as a contributor to this tutorial (in the metadata at the top of the file) :)

lybCNU commented:
@shiltemann Thank you for the thorough review and kind feedback

I've added myself to the CONTRIBUTORS.yaml file and updated the metadata in the tutorial as requested.

I also noticed that the Filter tool doesn't seem to work with column 16 (`c16`) in the `olympics.tsv` file. Could you please check if you're experiencing the same issue?

shiltemann commented:
@lybCNU I made some small formatting changes (split it into multiple hands-on boxes, put the general explanation outside the box), but looks great!

ok strangs @ column 16, I will check!

shiltemann commented on topics/introduction/tutorials/data-manipulation-olympics/tutorial.md:
```diff
@@ -570,6 +571,135 @@ This file contains a lot of data, but we may only be interested in a subset of t
+> 2. {% icon galaxy-eye %} **View** the filtered file.
+>
+> 3. **Rename** {% icon galaxy-pencil %} both outputs to something descriptive.
+>
```
could you add a small Q&A box at the end of each hands-on box where people can check their answers? So here it would just be?

Q: How many rows contained the string "Summer" in column 11?
A: <number of columns>

lybCNU replied:
I've added the small Q&A box at the end of each hands-on box as you suggested. Thank you for the great idea!

shiltemann replied:
Thanks!

shiltemann commented:
@lybCNU ok, I can confirm I have the same issues with filtering on c16

not exactly sure what's going on yet, but based on the stdout it is not parsing the lines correctly and labelling them as invalid:

```
Filtering with c16.find('Doubles') != -1, 
kept 0.00% of 234523 valid lines (234523 total lines).
Skipped 234522 invalid line(s) starting at line #2: "1	Jean-François Blanchy	M	1886	12 December 	Bordeaux, Gironde (FRA)	NA	NA	France	FRA	1912 Summer Olympics	1912	Summer	Stockholm	Tennis	Doubles, Men	NA"
```

might be a bug in the tool, I will look into it, thanks for reporting it!!

shiltemann commented:
Thanks again @lybCNU! :tada:
